hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ECUnderReplicationHandler.java

```diff
@@ -186,15 +191,17 @@ public Map<DatanodeDetails, SCMCommand<?>> processAndCreateCommands(
       Set<Integer> decomIndexes = replicaCount.decommissioningOnlyIndexes(true);
       if (decomIndexes.size() > 0) {
         final List<DatanodeDetails> selectedDatanodes =
-            getTargetDatanodes(replicas, container, decomIndexes.size());
+            getTargetDatanodes(excludedNodes, container, decomIndexes.size());
+        excludedNodes.addAll(selectedDatanodes);
```

Comment:
Nit: Is this adding is useful only for logging? After decommission indexes processing, we don't use that exclude list anymore except the logging? If so, we could add just before logging?

Reply:
There are two reasons I added it here - (1) to keep it consistent with the earlier code and (2) because we need to extend this class for maintenance mode, and we will need to add it here then anyway, and its something that could easily get missed. Therefore I think its best to keep it as it is.

Reply:
ok got it. Thanks for the reasoning.